src/resource_clients/request_queue.ts

```diff
@@ -234,6 +342,24 @@ export interface RequestQueueClientAddRequestResult {
     wasAlreadyHandled: boolean;
 }
 
+interface ProcessedRequest {
+    uniqueKey: string;
+    requestId: string;
+    wasAlreadyPresent: boolean;
+    wasAlreadyHandled: boolean;
+}
+
+interface UnprocessedRequest {
+    uniqueKey: string;
+    url: string;
+    method?: AllowedHttpMethods;
+}
```

Comment:
it feels a bit weird to have two separate interfaces for this, I also don't like the `UnprocessedRequest` name very much - it is a regular request, right? and the processed one is its extension, or not?

```ts
interface ApifyRequest {
    uniqueKey: string;
    url: string;
    method?: AllowedHttpMethods;
}

interface ProcessedRequest extends ApifyRequest {
    requestId: string;
    wasAlreadyPresent: boolean;
    wasAlreadyHandled: boolean;
}
```

seeing the rest of the interfaces here, many others should extend such base request type. it does not have to happen in this PR, just an idea for future.

Reply:
@B4nan Not really, they just share `uniqueKey` and not the other fields. But we might return those too from API in the future. Is it ok just to add a comment that we should revisit the shapes before we switch from experimental mode?